sdk/core/Azure.Core/src/Request.cs

```diff
@@ -73,6 +73,8 @@ public virtual string ClientRequestId
         /// </summary>
         public new RequestHeaders Headers => new(this);
 
+        internal TimeSpan NetworkTimeout { get; set; }
```

Comment:
Since Azure.Core's HttpPipeline has both CreateRequest and CreateMessage, we need a way for the transport to smuggle the network timeout from ClientOptions onto the message returned from the pipeline even when only CreateRequest is called.

Reply:
Actually, I think we should be able to remove this one.